packages/preferences/src/common/preference-service.ts

```diff
@@ -32,60 +34,70 @@ export class PreferenceService implements Disposable {
         this.toDispose.dispose();
     }
 
+    ready(): Promise<void> {
+        return this.isReady;
+    }
+
     protected onDidChangePreference(event: PreferenceChangedEvent): void {
+        for (const prefChange of event.changes) {
+            // Pref removed
```

Comment:
if you think that it is hard to reason about the following code, please refactor to make it readable, but I think it is readable as now and the comment can be just removed